Tests/KituraTests/TestServer.swift

```diff
@@ -149,4 +150,57 @@ class TestServer: KituraTest {
             XCTFail("Unexpected error starting server: \(error)")
         }
     }
+
+    func testServerRestart() {
+        let port = httpPort
+        let path = "/testServerRestart"
+        let body = "Server is running."
+
+        let router = Router()
+        router.get(path) { _, response, next in
+            response.send(body)
+            next()
+        }
+        Kitura.addHTTPServer(onPort: port, with: router)
+
+        let server = Kitura.httpServersAndPorts[0].server
+        server.sslConfig = KituraTest.sslConfig.config
+
+        let stopped = DispatchSemaphore(value: 0)
+        server.stopped {
+            stopped.signal()
+        }
+
+        testResponse(port: port, path: path, expectedBody: nil, expectedStatus: nil)
+
+        Kitura.start()
+        testResponse(port: port, path: path, expectedBody: body)
+        Kitura.stop(unregister: false)
+        stopped.wait()
```

Comment:
testResponse(port: port, path: path, expectedBody: body) only tests that are running. It doesn't that they exist altogether